packages/sdk/src/modules/Vaults.ts

```diff
@@ -0,0 +1,225 @@
+                (ass) => ass.underlying === asset
+              );
+
+              let extraDocs: string | undefined;
+
+              if (_asset) {
+                symbol = _asset.symbol;
+                decimals = _asset.decimals;
+                extraDocs = _asset.extraDocs;
+              }
+
+              const _adapters =
+                adapterCount > 0
+                  ? await Promise.all(
+                      Array.from(Array(adapterCount).keys()).map((i) => optimizedAPRVault.callStatic.adapters(i))
+                    )
+                  : [];
+
+              const adapters = await Promise.all(
+                _adapters.map(async (adapter) => {
+                  const adapterInstance = this.createCompoundMarketERC4626(adapter.adapter);
+                  const marketAddress = await adapterInstance.callStatic.market();
+                  const cTokenInstance = this.createCTokenWithExtensions(marketAddress);
+                  const comptrollerAddress = await cTokenInstance.callStatic.comptroller();
+
+                  return {
+                    adapter: adapter.adapter,
+                    allocation: adapter.allocation,
+                    market: marketAddress,
+                    comptroller: comptrollerAddress,
+                  };
+                })
+              );
+
+              const underlyingPrice = await mpo.callStatic.price(asset);
+              const totalSupplyNative =
+                Number(utils.formatUnits(totalSupply, decimals)) * Number(utils.formatUnits(underlyingPrice, 18));
+
+              return {
+                vault,
+                chainId: this.chainId,
+                totalSupply,
+                totalSupplyNative,
+                asset,
+                symbol,
+                supplyApy,
+                adapterCount,
+                emergencyExit,
+                adapters,
+                decimals,
+                underlyingPrice,
+                extraDocs,
+                performanceFee,
+                depositFee,
+                withdrawalFee,
+                managementFee,
+              };
+            })
+          );
+        } catch (error) {
+          this.logger.error(`get vaults error in chain ${this.chainId}:  ${error}`);
+
+          throw Error(
+            `Getting vaults failed in chain ${this.chainId}: ` + (error instanceof Error ? error.message : error)
+          );
+        }
+      } else {
+        return [];
+      }
+    }
+
+    async vaultApprove(vault: string, asset: string) {
+      const token = getContract(asset, EIP20InterfaceABI, this.signer);
+      const tx = await token.approve(vault, constants.MaxUint256);
+
+      return tx;
+    }
+
+    async convertToShares(amount: BigNumber, vault: string) {
+      const optimizedAPRVault = this.createOptimizedAPRVault(vault);
+
+      const totalSupply = await optimizedAPRVault.callStatic.totalSupply();
+      const totalAssets = await optimizedAPRVault.callStatic.totalAssets();
+
```

Comment:
it looks wrong to me to implement your own `convertToShares` and fee shares calculation.. can you use the `previewDeposit` and `previewWithdraw` fns instead?